tools/install.sh

```diff
@@ -102,7 +103,8 @@ setup_ohmyzsh() {
     exit 1
   }
 
-  if [ "$OSTYPE" = cygwin ] && git --version | grep -q msysgit; then
+  ostype=$(uname)
```

Comment:
Does it complain without this? `$OSTYPE` is guaranteed to exist in cygwin which is the only reason we're doing this.

Reply:
```sh
  if [ "$OSTYPE" = cygwin ] && git --version | grep -q msysgit; then
```
> `        ^-----^ SC2039: In POSIX sh, OSTYPE is undefined.`

Since the shebang is pointing to POSIX sh, `OSTYPE` cannot be used:
```sh
#!/bin/sh
```

About detecting Cygwin because it has this OSTYPE available, it has because its default sh shell points to bash. But since `ostype=$(uname)` is a portable POSIX compliant replacement, no reason to not do this instead.

Reply:
So it'll be an empty string and the condition will fail. Is it really worth it for the extra fork?

Reply:
Actually don't reply to that, it doesn't matter. Ready to merge isn't it?

Reply:
If forking was in a loop, I'd worry about performance. Otherwise, shell is not a general programming language but a good choice for portable installation scripts.

Yes ready to merge from my point, but I have no say, except that:
I tested my version of the script locally on a Linux Ubuntu box and it did not show any issue.
Now I did not test it on anything else, not even exotic BSD or antique SunOS or Xenix, or Minix ^^